x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -202,6 +202,17 @@ public class ApiKeyService {
         Property.NodeScope
     );
 
+    /**
+     * This setting is never registered by the security plugin - in order to configure strict request validation
+     * another plugin must register it as a boolean setting and override the value.
+     */
+    public static final Setting<Boolean> STRICT_REQUEST_VALIDATION_SETTING = Setting.boolSetting(
+        "xpack.security.authc.api_key.strict_request_validation.enabled",
+        false, // TODO : This will become true at a later time
+        Setting.Property.Dynamic,
```

Comment:
Was reviewing related changes and wondering if this should be `Setting.Property.OperatorDynamic` instead?

Reply:
Ah wait, that shouldn't be here at all. Thanks for catching this. 

I didn't even know about `OperatorDynamic` -- I _think_ it makes sense to switch it to that if it does what the name implies (though, based on tests, non-operators cannot call the cluster settings API at all).

Reply:
You are right. The setting has to be marked with `ServerlessPublic` to be accessible by non-operators. So it makes no difference in this case since we don't expose it in stateful.

Reply:
I'll use `OperatorDynamic` since it feels more future-proof and less confusing to anyone who knows about this annotation. Thanks @slobodanadamovic!